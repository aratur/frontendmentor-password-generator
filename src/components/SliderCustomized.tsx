import React from 'react';
import Slider from '@mui/material/Slider';
import { Box } from '@mui/material';

type Props = {
  value: number;
  max: number;
  handleChange: (e: Event) => void;
};

const iOSBoxShadow =
  '0 3px 1px rgba(0,0,0,0.1),0 4px 8px rgba(0,0,0,0.13),0 0 0 1px rgba(0,0,0,0.02)';

const SliderCustomized = (props: Props) => {
  const { value, max, handleChange } = props;

  return (
    <Box sx={{ marginInline: 2, marginTop: 0 }}>
      <Slider
        value={value}
        max={max}
        min={0}
        sx={{
          borderRadius: 0,
          '& .MuiSlider-track': {
            borderRadius: 0,
            height: '8px',
          },
          '& .MuiSlider-rail': {
            opacity: 1,
            backgroundColor: 'secondary.dark',
            height: '8px',
          },
          '& .MuiSlider-thumb': {
            height: 28,
            width: 28,
            backgroundColor: '#fff',
            transition: 'background-color 100ms ease-in-out',
            borderWidth: '2px',
            borderStyle: 'solid',
            borderColor: 'secondary.light',
            boxShadow: iOSBoxShadow,
            '&:focus, &:hover, &.Mui-active': {
              boxShadow:
                '0 3px 1px rgba(0,0,0,0.1),0 4px 8px rgba(0,0,0,0.3),0 0 0 1px rgba(0,0,0,0.02)',
              // Reset on touch devices, it doesn't add specificity
              borderWidth: '2px',
              borderStyle: 'solid',
              borderColor: 'primary.main',
              backgroundColor: 'secondary.dark',
              '@media (hover: none)': {
                boxShadow: iOSBoxShadow,
              },
            },
          },
        }}
        color="secondary"
        onChange={(e) => handleChange(e)}
      />
    </Box>
  );
};

export default SliderCustomized;
