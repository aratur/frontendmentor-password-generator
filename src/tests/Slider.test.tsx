import React from 'react';
import { screen, render } from '@testing-library/react';
import SliderCustomized from '../components/SliderCustomized';

describe('tests Slider', () => {
  it('renders with a slider', () => {
    const handleChange = vi.fn();
    render(<SliderCustomized handleChange={handleChange} max={10} value={1} />);
    const slider = screen.getByRole('slider');
    expect(slider).toBeInTheDocument();
  });
  it('matches snapshot', () => {
    const handleChange = vi.fn();
    const { container } = render(
      <SliderCustomized handleChange={handleChange} max={10} value={1} />
    );
    expect(container).toMatchSnapshot();
  });
});
