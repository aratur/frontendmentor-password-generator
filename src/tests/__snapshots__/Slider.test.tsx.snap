// Vitest Snapshot v1

exports[`tests Slider > matches snapshot 1`] = `
<div>
  <span
    class="MuiSlider-root MuiSlider-colorSecondary MuiSlider-sizeMedium css-5ejlq0-MuiSlider-root"
  >
    <span
      class="MuiSlider-rail css-14pt78w-MuiSlider-rail"
    />
    <span
      class="MuiSlider-track css-1gv0vcd-MuiSlider-track"
      style="left: 0%; width: 10%;"
    />
    <span
      class="MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorSecondary css-frv3qg-MuiSlider-thumb"
      data-focusvisible="false"
      data-index="0"
      style="left: 10%;"
    >
      <input
        aria-orientation="horizontal"
        aria-valuemax="10"
        aria-valuemin="0"
        aria-valuenow="1"
        data-index="0"
        max="10"
        min="0"
        step="1"
        style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
        type="range"
        value="1"
      />
    </span>
  </span>
</div>
`;
